---
title: "WP3: Data preparation in BiB"
author: "Kayleigh Easey and Gemma Sharp"
date: "09/03/2020"
output:
  rmdformats::material
---
# Introduction {.tabset}
```{r, echo=FALSE}
knitr::opts_chunk$set(error = TRUE)
```

## Description {.tabset}
This is the annotated code to prepare the BiB dataset for use in EPoCH. To run this code, copy and paste the relevant sections. 
All the code in this file (without the documentation) can be extracted using:
```{r eval=FALSE, include=TRUE}
rm(list=ls())
library(knitr)
purl("data_prep_bib.Rmd")
```

## Load packages {.tabset}
We are using tidyverse, which is a set of packages that work in harmony. The tidyverse package is designed to make it easy to install and load core packages from the tidyverse in a single command.

```{r eval=FALSE, include=TRUE}
packages <- c("dplyr", "purrr", "data.table","tidyverse","tinytex","rmarkdown")
lapply(packages, require, character.only=T)
```

## Linking BiB trios {.tabset}
* PregnancyID is uniquely assigned to each mother at recruitment. If the same mother returns to enrol for a second or third pregnancy, she gets a new PregnancyID. This is similar to the ALN in ALSPAC. Children and partners can be linked to the mother by the PregnancyID.
* ChildID is uniquely assigned to each child at birth. ChildID will be blank if the birth outcome is unknown. Use MotherID with ChildID to link siblings together. Note that twins have separate ChildIDs but the same PregnancyID.
* FatherID is uniquely assigned to partners post-recruitment. Use FatherID with PregnancyID to link fathers across pregnancies. Where FatherID matches across two pregnancies, but with different MotherIDs, this is a father with two pregnancies in the cohort with different mothers.
* MotherID is uniquely assigned to each mother post-recruitment. MotherID should be used when looking for pregnancies or children associated with the same mother. Where MotherID matches across two pregnancies, but with different FatherIDs, this is a mother with two pregnancies in the cohort with different fathers.

IMPORTANT - BiB has both Mother ID and Pregnancy ID - because many of the women had more than one pregnacy. So you need to decide if you want to operate on the Mother level (MotherID) OR pregnancy (PregnancyID) - this will affect how you combine your data 

```{r eval=FALSE, include=TRUE}
#Read in data
fathers_baseline <- get(load("/Volumes/MRC-IEU-research/data/bib/_latest/BIB_Phenotypes/data/rdata/Father_Baseline_Questionnaire_v7_fbqall_Data.Rdata"))
mothers_baseline <- get(load("/Volumes/MRC-IEU-research/data/bib/_latest/BIB_Phenotypes/data/rdata/Mother_Baseline_Questionnaire_v7_mbqall_Data.Rdata"))

preg_birth_eclprg <- get(load("/Volumes/MRC-IEU-research/data/bib/_latest/BIB_Phenotypes/data/rdata/Pregnancy_and_Birth_v7_eclprg_Data.Rdata"))
preg_birth_eclbby <- get(load("/Volumes/MRC-IEU-research/data/bib/_latest/BIB_Phenotypes/data/rdata/Pregnancy_and_Birth_v7_eclbby_Data.Rdata"))
preg_birth_prninf <- get(load("/Volumes/MRC-IEU-research/data/bib/_latest/BIB_Phenotypes/data/rdata/Pregnancy_and_Birth_v7_prninf_Data.Rdata"))
preg_birth_prnprg <- get(load("/Volumes/MRC-IEU-research/data/bib/_latest/BIB_Phenotypes/data/rdata/Pregnancy_and_Birth_v7_prnprg_Data.Rdata"))

child_follow_up_b6mtab <- get(load("/Volumes/MRC-IEU-research/data/bib/_latest/BIB_Phenotypes/data/rdata/Follow-up_v7_b6mtab_Data.Rdata"))
child_follow_up_b12tab <- get(load("/Volumes/MRC-IEU-research/data/bib/_latest/BIB_Phenotypes/data/rdata/Follow-up_v7_b12tab_Data.Rdata"))
child_follow_up_b18tab <- get(load("/Volumes/MRC-IEU-research/data/bib/_latest/BIB_Phenotypes/data/rdata/Follow-up_v7_b18tab_Data.Rdata"))
child_follow_up_b24tab <- get(load("/Volumes/MRC-IEU-research/data/bib/_latest/BIB_Phenotypes/data/rdata/Follow-up_v7_b24tab_Data.Rdata"))
child_follow_up_b36tab <- get(load("/Volumes/MRC-IEU-research/data/bib/_latest/BIB_Phenotypes/data/rdata/Follow-up_v7_b36tab_Data.Rdata"))

child_follow_up_chgcom <- get(load("/Volumes/MRC-IEU-research/data/bib/_latest/BIB_Phenotypes/data/rdata/Follow-up_v7_chgcom_Data.Rdata"))
child_follow_up_schlbp <- get(load("/Volumes/MRC-IEU-research/data/bib/_latest/BIB_Phenotypes/data/rdata/Follow-up_v7_schlbp_Data.Rdata"))

# merge together raw data (note that we don't want loads of instances of MotherID and FatherID (and ChildID when we're adding child-level variables) so we take those out of the data frames as we merge them (all other variable names should be unique):

# First merge anything at the pregnancy level (data mums, dads and pregnancies) by PregnancyID
raw_dat_parents <- merge(mothers_baseline,fathers_baseline[,-which(colnames(fathers_baseline) %in% c("MotherID","FatherID"))],by="PregnancyID",all.x=TRUE,all.y=FALSE)
raw_dat_parents <- merge(raw_dat_parents,preg_birth_eclprg[,-which(colnames(preg_birth_eclprg) %in% c("MotherID","FatherID"))],by="PregnancyID",all.x=TRUE,all.y=FALSE)
raw_dat_parents <- merge(raw_dat_parents,preg_birth_prnprg[,-which(colnames(preg_birth_prnprg) %in% c("MotherID","FatherID"))],by="PregnancyID",all.x=TRUE,all.y=FALSE)

# Next merge anything at the child level (data on children) by ChildID
raw_dat_child <- merge(preg_birth_eclbby,preg_birth_prninf[,-which(colnames(preg_birth_prninf) %in% c("MotherID","FatherID","PregnancyID"))],by="ChildID",all=TRUE)
raw_dat_child <- merge(raw_dat_child,child_follow_up_b6mtab[,-which(colnames(child_follow_up_b6mtab) %in% c("MotherID","FatherID","PregnancyID","has_bib1kc"))],by="ChildID",all=TRUE)
raw_dat_child <- merge(raw_dat_child,child_follow_up_b12tab[,-which(colnames(child_follow_up_b12tab) %in% c("MotherID","FatherID","PregnancyID","has_bib1kc"))],by="ChildID",all=TRUE)
raw_dat_child <- merge(raw_dat_child,child_follow_up_b18tab[,-which(colnames(child_follow_up_b18tab) %in% c("MotherID","FatherID","PregnancyID","has_bib1kc"))],by="ChildID",all=TRUE)
raw_dat_child <- merge(raw_dat_child,child_follow_up_b24tab[,-which(colnames(child_follow_up_b24tab) %in% c("MotherID","FatherID","PregnancyID","has_bib1kc"))],by="ChildID",all=TRUE)
raw_dat_child <- merge(raw_dat_child,child_follow_up_b36tab[,-which(colnames(child_follow_up_b36tab) %in% c("MotherID","FatherID","PregnancyID","has_bib1kc"))],by="ChildID",all=TRUE)
raw_dat_child <- merge(raw_dat_child,child_follow_up_schlbp[,-which(colnames(child_follow_up_schlbp) %in% c("MotherID","FatherID","PregnancyID","has_bib1kc"))],by="ChildID",all=TRUE)

# The dataset child_follow_up_chgcom is in long format and we want it in wide format. Also it sometimes has multiple observations per child at each timepoint, so we need to take an average of those. 
# I adapted this code from the code they are using in lifecycle to do this. First I made a function to take a variable, group it by child ID and age (in months) at growth measurement, calculate a mean for each child at each growth measurement month, and put it into wide format using the 'spread' function.
spread_chgcom <- function(variable){
Var <- child_follow_up_chgcom %>% group_by(ChildID, agecm_cgrowth) %>%
                             summarise(Value = mean(get(variable), na.rm = T))
Var <- Var %>% rename(X_ = agecm_cgrowth)
Var$Value[is.nan(Var$Value)] <- NA
Var <- Var %>% spread(X_, Value, sep = "")
colnames(Var) <- str_replace_all(colnames(Var), "X_", paste0(variable,"_"))
Var
}
# Then I just applied this function to all the variables we're interested in from this file and combined them all together into one dataset in wide format.
child_follow_up_chgcom_wide <- lapply(c("cabdo","cbmi","chead","cheight","cweight","czbmiuk90","czbmiwho06","czweiwho06"),spread_chgcom)
child_follow_up_chgcom_wide <- bind_cols(child_follow_up_chgcom_wide)
child_follow_up_chgcom_wide <- child_follow_up_chgcom_wide[-grep("ChildID",names(child_follow_up_chgcom_wide))[-1]]
# Finally, I took out any variables that are completely NA (i.e. if no children were measured in a particular month)
child_follow_up_chgcom_wide <- child_follow_up_chgcom_wide[colSums(!is.na(child_follow_up_chgcom_wide)) > 0]

# This could then be merged with the other child-level data:
raw_dat_child <- merge(raw_dat_child,child_follow_up_chgcom_wide,by.x="ChildID",by.y=colnames(child_follow_up_chgcom_wide)[grep(pattern="ChildID",colnames(child_follow_up_chgcom_wide))],all=TRUE)

# merge pregnancy level data with child level data to get a dataset at the child level:
## first get rid of any children whose mothers didn't complete the baseline questionnaire for their pregnancies (2613)
raw_dat_child <- raw_dat_child[raw_dat_child$PregnancyID %in% raw_dat_parents$PregnancyID,]
## merge to the child level, but keep any parents who didn't have child data (keep all in raw_dat_parents)
raw_dat <- merge(raw_dat_parents,raw_dat_child[,-which(colnames(raw_dat_child) %in% c("MotherID","FatherID","has_bib1kc"))],by="PregnancyID",all.x=TRUE,all.y=FALSE)

#set up new dat object:
dat<-raw_dat[,c("MotherID","FatherID","PregnancyID","ChildID")]

```

# Useful information {.tabset}

```{r eval=FALSE, include=TRUE}
dat$live_birth<-NA
dat$live_birth[raw_dat$prnbibirthoutc=="Live birth"]<-1
dat$live_birth[raw_dat$prnbibirthoutc%in%c("Miscarriage","Still birth")]<-0

# creating paternal participation var in BiB
dat$paternal_participation<-NA
dat$paternal_participation<-ifelse(is.na(dat$FatherID),0,1)
```

# Parents' smoking {.tabset}
## Ideal variables
Ideal harmonised smoking exposures for both parents are:

*	Any time up to birth: ever vs never smoker: binary variable
*	Early-onset: Smoking before age 11 vs no smoking before age 11
*	Pre-conception: Any vs no smoking in the three months prior to pregnancy
*	Pre-conception: Number of cigarettes per day in the three months prior to pregnancy (ordered categorical)
*	Pregnancy: Any smoking vs no smoking at any time during pregnancy
*	Pregnancy: Any smoking vs no smoking in the first trimester
*	Pregnancy: Any smoking vs no smoking in the second trimester
*	Pregnancy: Any smoking vs no smoking in the third trimester
*	Pregnancy: Number of cigarettes per day in the first trimester (ordered categorical)
*	Pregnancy: Number of cigarettes per day in the second trimester (ordered categorical)
*	Pregnancy: Number of cigarettes per day in the third trimester (ordered categorical)
* Passive smoke during pregnancy: binary variable
* Postnatal: ordered categorical up until age 2 (in line with the definition of the first 1000 days)
* Postnatal: any vs no smoking up to age 2

## Derive clean variables

**Maternal **

Any time up to birth: ever vs never smoker: binary variable 
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_ever_life_binary<-NA
dat$smoking_mother_ever_life_binary[which(raw_dat$smk0nowsmk=="No" | raw_dat$smk0regsmk=="No")]<-0
dat$smoking_mother_ever_life_binary[which(raw_dat$smk0nowsmk=="Yes" | raw_dat$smk0regsmk%in%c("Yes, more than 1 year","Yes, less than 1 year","Yes, not specified"))]<-1
```

Early-onset: Smoking before age 11 vs no smoking before age 11
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_before11_binary<-NA
dat$smoking_mother_before11_binary[raw_dat$smk0agestr<=11]<-1
dat$smoking_mother_before11_binary[raw_dat$smk0agestr>11]<-0
```

Pre-conception: Any vs no smoking in the three months prior to pregnancy
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_preconception_binary<-NA
dat$smoking_mother_preconception_binary[raw_dat$smk03mthb4=="None" | raw_dat$smk0regsmk=="No"]<-0
dat$smoking_mother_preconception_binary[raw_dat$smk03mthb4%in%c("1-5 a day","6-10 a day","11-20 a day","Over 20 a day")]<-1
```

Pre-conception: Number of cigarettes per day in the three months prior to pregnancy (ordered categorical)
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_preconception_ordinal<-NA
dat$smoking_mother_preconception_ordinal[raw_dat$smk03mthb4=="None"| raw_dat$smk0regsmk=="No"]<-"None"
dat$smoking_mother_preconception_ordinal[raw_dat$smk03mthb4%in%c("1-5 a day","6-10 a day")]<-"Light" #i.e. <=10 but not 0
dat$smoking_mother_preconception_ordinal[raw_dat$smk03mthb4=="11-20 a day"]<-"Moderate" #i.e. >10, <=20
dat$smoking_mother_preconception_ordinal[raw_dat$smk03mthb4=="Over 20 a day"]<-"Heavy" #i.e. >20
dat$smoking_mother_preconception_ordinal<-factor(dat$smoking_mother_preconception_ordinal, levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

Pregnancy: Number of cigarettes per day in the first trimester (ordered categorical)
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_firsttrim_ordinal<-NA
dat$smoking_mother_firsttrim_ordinal[raw_dat$smk0fr3mth=="None"| raw_dat$smk0regsmk=="No"]<-"None"
dat$smoking_mother_firsttrim_ordinal[raw_dat$smk0fr3mth%in%c("1-5 a day","6-10 a day")]<-"Light" #i.e. <=10 but not 0
dat$smoking_mother_firsttrim_ordinal[raw_dat$smk0fr3mth=="11-20 a day"]<-"Moderate" #i.e. >10, <=20
dat$smoking_mother_firsttrim_ordinal[raw_dat$smk0fr3mth=="Over 20 a day"]<-"Heavy" #i.e. >20
dat$smoking_mother_firsttrim_ordinal<-factor(dat$smoking_mother_firsttrim_ordinal, levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

Pregnancy: Any smoking vs no smoking in the first trimester (up to end of 12 weeks)
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_firsttrim_binary <- NA
dat$smoking_mother_firsttrim_binary[raw_dat$smk0fr3mth=="None"| raw_dat$smk0regsmk=="No"]<-0
dat$smoking_mother_firsttrim_binary[raw_dat$smk0fr3mth%in%c("1-5 a day","6-10 a day","11-20 a day","Over 20 a day")]<-1
```

Pregnancy: Any smoking vs no smoking in the second trimester
(The second are third timester variables are identical, due to how the question was asked "since 4th month of pregnancy")
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_secondtrim_binary<-NA
dat$smoking_mother_secondtrim_binary[raw_dat$smk04thmth=="None"| raw_dat$smk0regsmk=="No"]<-0
dat$smoking_mother_secondtrim_binary[raw_dat$smk04thmth%in%c("1-5 a day","6-10 a day","11-20 a day","Over 20 a day")]<-1
```

Pregnancy: Any smoking vs no smoking in the third trimester
(The second are third timester variables are identical, due to how the question was asked "since 4th month of pregnancy")
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_thirdtrim_binary<-NA
dat$smoking_mother_thirdtrim_binary[raw_dat$smk04thmth=="None"| raw_dat$smk0regsmk=="No"]<-0
dat$smoking_mother_thirdtrim_binary[raw_dat$smk04thmth%in%c("1-5 a day","6-10 a day","11-20 a day","Over 20 a day")]<-1
```

Pregnancy: Number of cigarettes per day in second trimester (ordered categorical)
(The second are third timester variables are identical, due to how th equestion was asked "since 4th month of pregnancy")

```{r eval=FALSE, include=TRUE}
dat$smoking_mother_secondtrim_ordinal<-NA
dat$smoking_mother_secondtrim_ordinal[raw_dat$smk04thmth=="None"| raw_dat$smk0regsmk=="No"]<-"None"
dat$smoking_mother_secondtrim_ordinal[raw_dat$smk04thmth%in%c("1-5 a day","6-10 a day")]<-"Light" #i.e. <=10 but not 0
dat$smoking_mother_secondtrim_ordinal[raw_dat$smk04thmth=="11-20 a day"]<-"Moderate" #i.e. >10, <=20
dat$smoking_mother_secondtrim_ordinal[raw_dat$smk04thmth=="Over 20 a day"]<-"Heavy" #i.e. >20
dat$smoking_mother_secondtrim_ordinal<-factor(dat$smoking_mother_secondtrim_ordinal, levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

Pregnancy: Number of cigarettes per day since in third trimester (ordered categorical)
(The second are third timester variables are identical, due to how th equestion was asked "since 4th month of pregnancy")
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_thirdtrim_ordinal<-NA
dat$smoking_mother_thirdtrim_ordinal[raw_dat$smk04thmth=="None"| raw_dat$smk0regsmk=="No"]<-"None"
dat$smoking_mother_thirdtrim_ordinal[raw_dat$smk04thmth%in%c("1-5 a day","6-10 a day")]<-"Light" #i.e. <=10 but not 0
dat$smoking_mother_thirdtrim_ordinal[raw_dat$smk04thmth=="11-20 a day"]<-"Moderate" #i.e. >10, <=20
dat$smoking_mother_thirdtrim_ordinal[raw_dat$smk04thmth=="Over 20 a day"]<-"Heavy" #i.e. >20
dat$smoking_mother_thirdtrim_ordinal<-factor(dat$smoking_mother_thirdtrim_ordinal, levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

Pregnancy: Any smoking vs no smoking at any time during pregnancy
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_ever_pregnancy_binary<-NA
dat$smoking_mother_ever_pregnancy_binary[raw_dat$smk0smkprg=="Yes"]<-1
dat$smoking_mother_ever_pregnancy_binary[raw_dat$smk0smkprg=="No"]<-0
```

Passive smoke during pregnancy: binary variable
```{r eval=FALSE, include=TRUE}
dat$smoking_mother_passive_binary<-NA
dat$smoking_mother_passive_binary[raw_dat$smk0expcig%in%c("Yes","Less than 1 hour per day")]<-1
dat$smoking_mother_passive_binary[raw_dat$smk0expcig=="No"]<-0
```

Postnatal smoking is not available

**Paternal**

Only 2 relevant smoking questions were asked to fathers, current smoker (fbqsmokes), and number of cigarettes per day (fbqsmokesamt). These have been used to create the below variables, some of which are duplicated for ever in pregnancy or third trimester due to being asked at 28 weeks.

Any smoking vs no smoking in the third trimester 
```{r eval=FALSE, include=TRUE}
dat$smoking_father_thirdtrim_binary<-NA
dat$smoking_father_thirdtrim_binary[raw_dat$fbqsmokes=="No"]<-0
dat$smoking_father_thirdtrim_binary[raw_dat$fbqsmokes=="Yes"]<-1
```

Pregnancy: Number of cigarettes per day in third trimester (ordered categorical)
```{r eval=FALSE, include=TRUE}
dat$smoking_father_thirdtrim_ordinal<-NA
dat$smoking_father_thirdtrim_ordinal[raw_dat$fbqsmokes=="No"]<-"None"
dat$smoking_father_thirdtrim_ordinal[raw_dat$fbqsmokesamt%in%c("Less than 1 a day","1-5","6-10")]<-"Light" #i.e. <=10 but not 0
dat$smoking_father_thirdtrim_ordinal[raw_dat$fbqsmokesamt=="11-20"]<-"Moderate" #i.e. >10, <=20
dat$smoking_father_thirdtrim_ordinal[raw_dat$fbqsmokesamt=="Over 20"]<- "Heavy" #i.e. >20
dat$smoking_father_thirdtrim_ordinal<-factor(dat$smoking_father_thirdtrim_ordinal, levels=c("None","Light","Moderate","Heavy"),ordered=T)
```

Pregnancy: Any smoking vs no smoking at any time during pregnancy
(This is identical to the smoking_father_thirdtrim_binary question)
```{r eval=FALSE, include=TRUE}
dat$smoking_father_ever_pregnancy_binary<-NA
dat$smoking_father_ever_pregnancy_binary[raw_dat$fbqsmokes=="No"]<-0
dat$smoking_father_ever_pregnancy_binary[raw_dat$fbqsmokes=="Yes"]<-1
```

# Parents' alcohol {.tabset}

## Ideal variables
Ideal harmonised alcohol exposures for both parents are:

*	Pre-conception: Any vs no drinking in the three months prior to pregnancy
*	Pre-conception: Number of units per day in the three months prior to pregnancy (ordered categorical)
*	Pregnancy: Any drinking vs no drinking at any time during pregnancy
*	Pregnancy: Any drinking vs no drinking in the first trimester
*	Pregnancy: Any drinking vs no drinking in the second trimester
*	Pregnancy: Any drinking vs no drinking in the third trimester
*	Pregnancy: Number of units per day in the first trimester (ordered categorical)
*	Pregnancy: Number of units per day in the second trimester (ordered categorical)
*	Pregnancy: Number of units per day in the third trimester (ordered categorical)
* Binge drinking in first trimester (binary)
* Binge drinking in second trimester (binary)
* Binge drinking in third trimester (binary)
* Binge drinking at any point during pregnancy
* Postnatal alcohol use ordinal, up to 2 years  21 months (first 1000 days)
* Postnatal alcohol up to age 2 binary

## Derive clean variables {.tabset}

**Maternal **

Note that some women reported drinking in pregnancy but did not indicate how much, so numbers for the binary and ordinal variables do not necessarily match

Before pregnancy

```{r eval= FALSE, include= TRUE}
#Drank alcohol 3 months before pregnancy (binary)
dat$alcohol_mother_preconception_binary<-NA
dat$alcohol_mother_preconception_binary[raw_dat$alc0dr3mb4=="No"]<-0
dat$alcohol_mother_preconception_binary[raw_dat$alc0dr3mb4%in%c("Yes, once a week", "Yes, occasionally","Yes, not specified" )]<-1

#average units beer per week 3 months before pregnancy
raw_dat$alcohol_mother_beer_pw_preconception_cont<-raw_dat$alc0brav3b 
#average units other per week 3 months before pregnancy
raw_dat$alcohol_mother_other_pw_preconception_cont<-raw_dat$alc0oaav3b
#average units spirits per week 3 months before pregnancy
raw_dat$alcohol_mother_spirit_pw_preconception_cont<-raw_dat$alc0spav3b
#average units wine per week 3 months before pregnancy
raw_dat$alcohol_mother_wine_pw_preconception_cont<-raw_dat$alc0wnav3b

#total average units per week 3 months before pregnancy
raw_dat$alcohol_mother_preconception_continuous<-rowSums(raw_dat[,c("alcohol_mother_beer_pw_preconception_cont","alcohol_mother_other_pw_preconception_cont","alcohol_mother_spirit_pw_preconception_cont","alcohol_mother_wine_pw_preconception_cont")],na.rm=TRUE)
raw_dat$alcohol_mother_preconception_continuous[apply(raw_dat[,c("alcohol_mother_beer_pw_preconception_cont","alcohol_mother_other_pw_preconception_cont","alcohol_mother_spirit_pw_preconception_cont","alcohol_mother_wine_pw_preconception_cont")],1,function(x)all(is.na(x)))]<-NA

## To align with ALSPAC's coding, 1 glass of drink taken as 2 units (average)   
# Number of units per day in preconception (ordered categorical)
dat$alcohol_mother_preconception_ordinal <-NA
dat$alcohol_mother_preconception_ordinal[raw_dat$alcohol_mother_preconception_continuous==0|dat$alcohol_mother_preconception_binary==0]<-"None"
dat$alcohol_mother_preconception_ordinal[raw_dat$alcohol_mother_preconception_continuous>0 & raw_dat$alcohol_mother_preconception_continuous <=2]<-"Light" #<1 glass per week (<2 units)
dat$alcohol_mother_preconception_ordinal[raw_dat$alcohol_mother_preconception_continuous >2 & raw_dat$alcohol_mother_preconception_continuous <=13]<-"Moderate" #1+glasses pwk, but not daily (<13 units)
dat$alcohol_mother_preconception_ordinal[raw_dat$alcohol_mother_preconception_continuous >13]<-"Heavy" #daily drinking, 1-2pd, 3-9pd, 10+pd (>13 units)
dat$alcohol_mother_preconception_ordinal<-factor(dat$alcohol_mother_preconception_ordinal, levels=c("None", "Light","Moderate","Heavy"), ordered=T)
```

First trimester

```{r eval= FALSE, include= TRUE}

#Drank alcohol in first trimester (binary)
# Asked as first three months of pregnancy 
dat$alcohol_mother_firsttrim_binary<-NA
dat$alcohol_mother_firsttrim_binary[raw_dat$alc0drfr3m=="No"]<-0
dat$alcohol_mother_firsttrim_binary[raw_dat$alc0drfr3m%in%c("Yes, once a week","Yes, occasionally","Yes, not specified")]<-1

#average units beer first trimester
raw_dat$alcohol_mother_beer_firsttrim<-raw_dat$alc0bravf3
#average units other first trimester
raw_dat$alcohol_mother_other_firsttrim<-raw_dat$alc0oaavf3
#average units spirits first  trimester
raw_dat$alcohol_mother_spirits_firsttrim<-raw_dat$alc0spavf3
#average units wine first 3  trimester
raw_dat$alcohol_mother_wine_firsttrim<-raw_dat$alc0wnavf3

#total average units per week/day in first trimester
raw_dat$alcohol_mother_firsttrim_continuous<-rowSums(raw_dat[,c("alcohol_mother_beer_firsttrim","alcohol_mother_other_firsttrim","alcohol_mother_spirits_firsttrim","alcohol_mother_wine_firsttrim")],na.rm=TRUE)
raw_dat$alcohol_mother_firsttrim_continuous[apply(raw_dat[,c("alcohol_mother_beer_firsttrim","alcohol_mother_other_firsttrim","alcohol_mother_spirits_firsttrim","alcohol_mother_wine_firsttrim")],1,function(x)all(is.na(x)))]<-NA

# Number of units per week in the first trimester (ordered categorical)
## To align with ALSPAC's coding, 1 glass of drink taken as 2 units (average)   
dat$alcohol_mother_firsttrim_ordinal <-NA
dat$alcohol_mother_firsttrim_ordinal[raw_dat$alcohol_mother_firsttrim_continuous==0|dat$alcohol_mother_firsttrim_binary==0]<-"None"
dat$alcohol_mother_firsttrim_ordinal[raw_dat$alcohol_mother_firsttrim_continuous>0 & raw_dat$alcohol_mother_firsttrim_continuous <=2]<-"Light" #<1 glass per week
dat$alcohol_mother_firsttrim_ordinal[raw_dat$alcohol_mother_firsttrim_continuous >2 & raw_dat$alcohol_mother_firsttrim_continuous <=13]<-"Moderate" #1+glasses pwk, but not daily (12 units)
dat$alcohol_mother_firsttrim_ordinal[raw_dat$alcohol_mother_firsttrim_continuous >13]<-"Heavy" #daily drinking
dat$alcohol_mother_firsttrim_ordinal<-factor(dat$alcohol_mother_firsttrim_ordinal, levels=c("None", "Light","Moderate","Heavy"), ordered=T)
```

Second trimester
```{r eval= FALSE, include= TRUE}

#drank alcohol in second trimester (binary)
#(The second are third timester variables are identical, due to how the question was asked "since 4th month of pregnancy")
dat$alcohol_mother_secondtrim_binary<-NA
dat$alcohol_mother_secondtrim_binary[raw_dat$alc0dr4thm=="No"]<-0
dat$alcohol_mother_secondtrim_binary[raw_dat$alc0dr4thm%in%c("Yes, once a week","Yes, occasionally","Yes, not specified")]<-1

#average units beer since 4th month of pregnancy
raw_dat$alcohol_mother_beer_pw_since4thmonth_cont<-raw_dat$alc0brav4m
#average units other since 4th m
raw_dat$alcohol_mother_other_pw_since4thmonth_cont<-raw_dat$alc0oaav4m
#average units spirits since 4th m
raw_dat$alcohol_mother_spirits_pw_since4thmonth_cont<-raw_dat$alc0spav4m
#average units wine since 4th m
raw_dat$alcohol_mother_wine_pw_since4thmonth_cont<-raw_dat$alc0wnav4m

#total average units per week/day since fourth month of pregnancy
raw_dat$alcohol_mother_pw_since4thmonth_cont<-rowSums(raw_dat[,c("alcohol_mother_beer_pw_since4thmonth_cont","alcohol_mother_other_pw_since4thmonth_cont","alcohol_mother_spirits_pw_since4thmonth_cont","alcohol_mother_wine_pw_since4thmonth_cont")],na.rm=TRUE)
raw_dat$alcohol_mother_pw_since4thmonth_cont[apply(raw_dat[,c("alcohol_mother_beer_pw_since4thmonth_cont","alcohol_mother_other_pw_since4thmonth_cont","alcohol_mother_spirits_pw_since4thmonth_cont","alcohol_mother_wine_pw_since4thmonth_cont")],1,function(x)all(is.na(x)))]<-NA

# Number of units per day in second trimester (ordered categorical)
## To align with ALSPAC's coding, 1 glass of drink taken as 2 units (average)   
dat$alcohol_mother_secondtrim_ordinal <-NA
dat$alcohol_mother_secondtrim_ordinal[raw_dat$alcohol_mother_pw_since4thmonth_cont==0|dat$alcohol_mother_secondtrim_binary==0]<-"None"
dat$alcohol_mother_secondtrim_ordinal[raw_dat$alcohol_mother_pw_since4thmonth_cont>0 & raw_dat$alcohol_mother_pw_since4thmonth_cont <=2]<-"Light" #<1 glass per week
dat$alcohol_mother_secondtrim_ordinal[raw_dat$alcohol_mother_pw_since4thmonth_cont >2 & raw_dat$alcohol_mother_pw_since4thmonth_cont <=13]<-"Moderate" #1+glasses pwk, but not daily (12 units)
dat$alcohol_mother_secondtrim_ordinal[raw_dat$alcohol_mother_pw_since4thmonth_cont >13]<-"Heavy" #daily drinking
dat$alcohol_mother_secondtrim_ordinal<-factor(dat$alcohol_mother_secondtrim_ordinal, levels=c("None", "Light","Moderate","Heavy"), ordered=T)
```

Third trimester

```{r eval= FALSE, include= TRUE}
#NOTE: The second and third trimester variables are identical, due to how the question was asked "since 4th month of pregnancy"
dat$alcohol_mother_thirdtrim_ordinal <- dat$alcohol_mother_secondtrim_ordinal
dat$alcohol_mother_thirdtrim_binary <- dat$alcohol_mother_secondtrim_binary
```

Binge drinking (5 or more on any one occasion) - 2nd and 3rd trim are the same
```{r eval= FALSE, include= TRUE}
dat$alcohol_mother_binge1_binary <-NA
dat$alcohol_mother_binge1_binary[raw_dat$alc05ufr3m%in%c("Everyday","Nearly everyday","1-4 times/week","1-3 times a month","Rarely")]<-1
dat$alcohol_mother_binge1_binary[raw_dat$alc05ufr3m%in%c("Never")|raw_dat$alc0drfr3m=="No"]<-0

dat$alcohol_mother_binge2_binary <-NA
dat$alcohol_mother_binge2_binary[raw_dat$alc05u4thm%in%c("Everyday","Nearly everyday","1-4 times/week","1-3 times a month","Rarely")]<-1
dat$alcohol_mother_binge2_binary[raw_dat$alc05u4thm%in%c("Never")|raw_dat$alc0dr4thm=="No"]<-0

dat$alcohol_mother_binge3_binary <- dat$alcohol_mother_binge2_binary

dat$alcohol_mother_bingepreg_binary <- NA
dat$alcohol_mother_bingepreg_binary[dat$alcohol_mother_binge1_binary==0 & dat$alcohol_mother_binge2_binary==0] <-0
dat$alcohol_mother_bingepreg_binary[dat$alcohol_mother_binge1_binary==1 | dat$alcohol_mother_binge2_binary==1] <-1
```

Any time in pregnancy
```{r eval= FALSE, include= TRUE}
dat$alcohol_mother_ever_pregnancy_binary<-NA
dat$alcohol_mother_ever_pregnancy_binary[dat$alcohol_mother_firsttrim_binary==0 &dat$alcohol_mother_secondtrim_binary==0 & dat$alcohol_mother_thirdtrim_binary == 0] <-0
dat$alcohol_mother_ever_pregnancy_binary[dat$alcohol_mother_firsttrim_binary==1 |dat$alcohol_mother_secondtrim_binary ==1 |dat$alcohol_mother_thirdtrim_binary ==1] <-1
```

postnatal alcohol use is not available for mothers

**Paternal**

Only 2 partner variables were available for alcohol, do you drink alcohol (fbqalcohol) and weekly avg no of units (fbqalcohol). Data was collected at baseline:

```{r eval=FALSE, include=TRUE}
#Any drinking vs no drinking at any time during pregnancy
#To note, the question asked was 'do you drink alcohol', this therefore could have been answered for pre-pregnancy drinking.
dat$alcohol_father_ever_pregnancy_binary<-NA
dat$alcohol_father_ever_pregnancy_binary[raw_dat$fbqalcohol=="No"]<-0
dat$alcohol_father_ever_pregnancy_binary[raw_dat$fbqalcohol=="Yes"]<-1

#Number of units in third trimester per week (continuous)
raw_dat$alcohol_father_amt_preg_cont<-raw_dat$fbqalcoholamt

#Number of units per day in the third trimester (ordered categorical)
dat$alcohol_father_thirdtrim_ordinal <-NA
dat$alcohol_father_thirdtrim_ordinal[raw_dat$alcohol_father_amt_preg_cont==0|raw_dat$fbqalcohol=="No"]<-"None"
dat$alcohol_father_thirdtrim_ordinal[raw_dat$alcohol_father_amt_preg_cont>0 & raw_dat$alcohol_father_amt_preg_cont <=2]<-"Light" #<1 glass per week
dat$alcohol_father_thirdtrim_ordinal[raw_dat$alcohol_father_amt_preg_cont >2 & raw_dat$alcohol_father_amt_preg_cont <=13]<-"Moderate" #1+glasses pwk, but not daily (12 units)
dat$alcohol_father_thirdtrim_ordinal[raw_dat$alcohol_father_amt_preg_cont >13]<-"Heavy" #daily drinking
dat$alcohol_father_thirdtrim_ordinal<-factor(dat$alcohol_father_thirdtrim_ordinal, levels=c("None", "Light","Moderate","Heavy"), ordered=T)

#Any alcohol Vs none in third trimester
dat$alcohol_father_thirdtrim_binary<-dat$alcohol_father_ever_pregnancy_binary
```

# Parents' caffeine {.tabset}
## Ideal variables 
Caffeine from tea, coffee and cola only.

Ideal harmonised caffeine exposures for both parents are:

* Pre-conception: Any vs no caffeine in the three months prior to pregnancy
* Pre-conception: Amount of caffeine (mg/day) per day in the three months prior to pregnancy

* Pregnancy: caffeine from coffee in first trimester (mg/day)
* Pregnancy: caffeine from tea in first trimester (mg/day)
* Pregnancy: caffeine from cola in first trimester (mg/day)
* Pregnancy: total caffeine in first trimester (mg/day)

* Pregnancy: caffeine from coffee in second trimester (mg/day)
* Pregnancy: caffeine from tea in second trimester (mg/day)
* Pregnancy: caffeine from cola in second trimester (mg/day)
* Pregnancy: total caffeine in second trimester (mg/day)

* Pregnancy: caffeine from coffee in third trimester (mg/day)
* Pregnancy: caffeine from tea in third trimester (mg/day)
* Pregnancy: caffeine from cola in third trimester (mg/day)
* Pregnancy: total caffeine in third trimester (mg/day)

* Pregnancy: Any caffeine vs no caffeine in the first trimester
* Pregnancy: Any caffeine vs no caffeine in the second trimester
* Pregnancy: Any caffeine vs no caffeine in the third trimester

* Pregnancy: Average caffeine per day in pregnancy
* Pregnancy: Any caffeine vs no caffeine in pregnancy

* Postnatal: caffeine from coffee up to age 2 (mg/day)
* Postnatal: caffeine from tea up to age 2 (mg/day)
* Postnatal: caffeine from cola up to age 2 (mg/day)
* Postnatal: total caffeine up to age 2 (mg/day)
* Postnatal: any vs none

## Derive clean variables

**Maternal**

The only available maternal caffeine variables are Cups of caffeinated filter/cafetiere coffee per day (cdr0fccfpd), Cups of caffeinated herbal tea pd (cdr0htcfpd), Cups of caffeinated tea per day (cdr0tecfpd), Cups of caffeinated cola pd (cdr0clcfpd,) diet cola pd (cdr0dccfpd)... decided to omit caffeinated herbal tea, as it was omitted from ALSPAC (so the variable will be caffeine from black tea, coffee and cola). Diet cola has more caffeine than normal cola, but we have treated them both the same as in ALSPAC no distinction was made.

```{r eval=FALSE, include=TRUE}
# Cups of coffee per day, mg/pd

# filter
raw_dat$caffeine_mother_coffee_filter_thirdtrim_continuous<-raw_dat$cdr0dccfpd
# cups of coffee multiplied by 57 (mg caffeine in 1 cup of cofee)
raw_dat$caffeine_mother_coffee_filter_thirdtrim_continuous<-raw_dat$caffeine_mother_coffee_filter_thirdtrim_continuous *57 #tranform cups to mg per day


# instant
raw_dat$caffeine_mother_coffee_instant_thirdtrim_continuous<-raw_dat$cdr0iccfpd
# cups of coffee multiplied by 57 (mg caffeine in 1 cup of cofee)
raw_dat$caffeine_mother_coffee_instant_thirdtrim_continuous<-raw_dat$caffeine_mother_coffee_instant_thirdtrim_continuous *57 #tranform cups to mg per day

# total coffee
dat$caffeine_mother_coffee_thirdtrim_continuous<-NA
dat$caffeine_mother_coffee_thirdtrim_continuous<- rowSums(raw_dat[,c("caffeine_mother_coffee_filter_thirdtrim_continuous","caffeine_mother_coffee_instant_thirdtrim_continuous")],na.rm=TRUE)
dat$dat$caffeine_mother_coffee_thirdtrim_continuous[apply(raw_dat[,c("caffeine_mother_coffee_instant_thirdtrim_continuous","caffeine_mother_coffee_filter_thirdtrim_continuous")],1,function(x)all(is.na(x)))]<-NA


# Cups of caffeinated tea per day, mg/pd
dat$caffeine_mother_tea_thirdtrim_continuous<-raw_dat$cdr0tecfpd
# cups of tea multiplied by 27 (mg caffeine in 1 cup tea)
dat$caffeine_mother_tea_thirdtrim_continuous<-dat$caffeine_mother_tea_thirdtrim_continuous*27 #tranform cups to mg per day

# Cups of caffeinated herbal tea per day, mg/pd
#raw_dat$caffeine_mother_herbaltea_preg_cont<-raw_dat$cdr0htcfpd
# cups of herbal tea multiplied by 27 (mg caffeine in 1 cup tea)
#raw_dat$caffeine_mother_herbaltea_preg_cont<-raw_dat$caffeine_mother_herbaltea_preg_cont*27 #tranform cups to mg per day

# Add together standard tea and herbal tea (if we happy that the mg in herbal are roughly similar)
#dat$caffeine_mother_tea_thirdtrim_continuous<-rowSums(raw_dat[,c("caffeine_mother_blacktea_thirdtrim_cont","caffeine_mother_herbaltea_preg_cont")],na.rm=TRUE)

#Cups of caffeinated cola per day (notdiet)
raw_dat$caffeine_mother_cola_notdiet_thirdtrim_continuous<-raw_dat$cdr0clcfpd
# cups of coffee multiplied by 20 (mg caffeine in 1 cup of cola)
raw_dat$caffeine_mother_cola_notdiet_thirdtrim_continuous<-raw_dat$caffeine_mother_cola_notdiet_thirdtrim_continuous*20 #tranform cups to mg per day

# Cups of caffeinated cola mg/pd (diet)
raw_dat$caffeine_mother_cola_diet_preg_continuous<-raw_dat$cdr0dccfpd
# cups of coffee multiplied by 20 (mg caffeine in 1 cup of cola)
raw_dat$caffeine_mother_cola_diet_preg_continuous<-raw_dat$caffeine_mother_cola_diet_preg_continuous*20 #tranform cups to mg per day

# Total cups caffeinated cola mg/pd (diet and non-diet)
dat$caffeine_mother_cola_thirdtrim_continuous<- rowSums(raw_dat[,c("caffeine_mother_cola_notdiet_thirdtrim_continuous","caffeine_mother_cola_diet_preg_continuous")],na.rm=TRUE)
dat$caffeine_mother_cola_thirdtrim_continuous[apply(raw_dat[,c("caffeine_mother_cola_notdiet_thirdtrim_continuous","caffeine_mother_cola_diet_preg_continuous")],1,function(x)all(is.na(x)))]<-NA

# Total caffeine per day in third trimester mg/day (continuous)
dat$caffeine_mother_total_thirdtrim_continuous<-rowSums(dat[,c("caffeine_mother_coffee_thirdtrim_continuous","caffeine_mother_tea_thirdtrim_continuous","caffeine_mother_cola_thirdtrim_continuous")],na.rm=TRUE)
dat$caffeine_mother_total_thirdtrim_continuous[apply(dat[,c("caffeine_mother_coffee_thirdtrim_continuous","caffeine_mother_tea_thirdtrim_continuous","caffeine_mother_cola_thirdtrim_continuous")],1,function(x)all(is.na(x)))]<-NA

# Average caffeine mg/pd in pregnancy (continuous)
# This is a duplicated variable of 'caffeine_mother_tea_thirdtrim_continuous', as we only have data for the third trimester
dat$caffeine_mother_total_ever_pregnancy_continuous<-dat$caffeine_mother_total_thirdtrim_continuous

# Any caffeine Vs none in third trimester
dat$caffeine_mother_total_thirdtrim_binary<- NA
dat$caffeine_mother_total_thirdtrim_binary[dat$caffeine_mother_total_thirdtrim_continuous==0]<-0
dat$caffeine_mother_total_thirdtrim_binary[dat$caffeine_mother_total_thirdtrim_continuous>0]<-1

# Pregnancy: Any caffeine vs no caffeine in pregnancy
# Duplicate variable of 'caffeine_mother_total_thirdtrim_binary', as only have data for third trimester
dat$caffeine_mother_total_ever_pregnancy_binary <- dat$caffeine_mother_total_thirdtrim_binary
```


**Paternal**

There are no caffeine questions for fathers during pregnancy 


# Parents' physical activity {.tabset}
There are no physical activity questions for either parent during pregnancy 

# Child's anthropometry & adiposity {.tabset}

## Ideal variables

Ideal harmonised anthropometry and adiposity measures for offspring are:

*	Birthweight (g)
*	SGA
*	LGA
* Low BW <2500g vs >=2500g & <=4500g
* High BW > 4500g vs >=2500g & <=4500g
*	Birth length (cm)
*	Head circumference (cm)
*	Height (cm)
*	Waist circumference (cm)
*	Weight (kg)
*	BMI (kg/m2)
*	WHO categories of overweight/obese/underweight vs normal weight
*	Fat mass index

Ages are defined as:

* at birth or in first year of life (>0 to <1; 0-11m)
* childhood stage 1 (around age 2; >=1 to <3); 12-35m
* childhood stage 2 (around age 4; >=3 to <5; 36-59m)
* childhood stage 3 (around age 6; >=5 to <8;60-95m)
* childhood stage 4 (around age 9; >=8 to <11;96-143m)

In BiB anthro measures are taken from varying sources:

* eClipse_agecy_cgrowth
* BiB1000-06m_agecy_cgrowth 
* BiB1000-12m_agecy_cgrowth
* BiB1000-18m_agecy_cgrowth
* BiB1000-24m_agecy_cgrowth
* BiB1000-36m_agecy_cgrowth
* Primary Care_agecy_cgrowth
* Child Health Record_agecy_cgrowth
* NCMP_agecy_cgrowth
* School Nurse_agecy_cgrowth


## Derive clean variables {.tabset}
```{r eval=FALSE, include=TRUE}
# At birth:
# Birthweight
dat$anthro_birthweight <-raw_dat$eclbirthwt

# High BW > 4500g vs >=2500g & <=4500g 
dat$anthro_birthweight_high_binary <- NA
dat$anthro_birthweight_high_binary[dat$anthro_birthweight>4500]<-1
dat$anthro_birthweight_high_binary[dat$anthro_birthweight>=2500&dat$anthro_birthweight<=4500]<-0

#Low BW <2500g vs >=2500g & <=4500g
dat$anthro_birthweight_low_binary <- NA
dat$anthro_birthweight_low_binary[dat$anthro_birthweight<2500]<-1
dat$anthro_birthweight_low_binary[dat$anthro_birthweight>=2500&dat$anthro_birthweight<=4500]<-0

#	SGA
dat$anthro_birthweight_sga_binary <- NA
dat$anthro_birthweight_sga_binary[raw_dat$eclsgaukwho=="No"]<-0
dat$anthro_birthweight_sga_binary[raw_dat$eclsgaukwho=="Yes"]<-1
#	LGA
dat$anthro_birthweight_lga_binary <- NA
dat$anthro_birthweight_lga_binary[raw_dat$ecllgaukwho=="No"]<-0
dat$anthro_birthweight_lga_binary[raw_dat$ecllgaukwho=="Yes"]<-1

#Head circumference at birth (cm)
dat$anthro_head_circ_birth<-raw_dat$eclheadcir

#birth length (crown-heel) taken as height at birth
dat$anthro_crown_heel_birth <- raw_dat$cheight_0

# setting the month range for each stage
stagebirth_mintime <- 0
stagebirth_maxtime <- 11
stage1_mintime <- 12
stage1_maxtime <- 35
stage2_mintime <- 36
stage2_maxtime <-59
stage3_mintime <- 60
stage3_maxtime <- 95
stage4_mintime <- 96
stage4_maxtime <- 143

select_oldest_measure <- function(prefix,stage_min,stage_max){
  col_names <- paste0(prefix,stage_min:stage_max)
  df <- raw_dat[,which(colnames(raw_dat) %in% col_names)]
  oldest_measure <- apply(df,1,function(x) ifelse(all(is.na(x)),NA,tail(na.omit(x),1)))
  timepoint <- apply(df,1,function(x) ifelse(all(is.na(x)),NA,names(tail(na.omit(x),1))))
timepoint<-as.numeric(str_remove(timepoint,prefix))
list(oldest_measure,timepoint)
}

dat$anthro_height_stage1 <- select_oldest_measure("cheight_",stage1_mintime,stage1_maxtime)[[1]]
dat$covs_stage1_timepoint <- select_oldest_measure("cheight_",stage1_mintime,stage1_maxtime)[[2]]

dat$anthro_height_stage2 <- select_oldest_measure("cheight_",stage2_mintime,stage2_maxtime)[[1]]
dat$covs_stage2_timepoint <- select_oldest_measure("cheight_",stage2_mintime,stage2_maxtime)[[2]]

dat$anthro_height_stage3 <- select_oldest_measure("cheight_",stage3_mintime,stage3_maxtime)[[1]]
dat$covs_stage3_timepoint <- select_oldest_measure("cheight_",stage3_mintime,stage3_maxtime)[[2]]

dat$anthro_weight_stage1 <- select_oldest_measure("cweight_",stage1_mintime,stage1_maxtime)[[1]]
dat$covs_stage1_timepoint <- select_oldest_measure("cweight_",stage1_mintime,stage1_maxtime)[[2]]

dat$anthro_weight_stage2 <- select_oldest_measure("cweight_",stage2_mintime,stage2_maxtime)[[1]]
dat$covs_stage2_timepoint <- select_oldest_measure("cweight_",stage2_mintime,stage2_maxtime)[[2]]

dat$anthro_weight_stage3 <- select_oldest_measure("cweight_",stage3_mintime,stage3_maxtime)[[1]]
dat$covs_stage3_timepoint <- select_oldest_measure("cweight_",stage3_mintime,stage3_maxtime)[[2]]

dat$anthro_head_circ_stage1 <- select_oldest_measure("chead_",stage1_mintime,stage1_maxtime)[[1]]

dat$anthro_head_circ_stage2 <- select_oldest_measure("chead_",stage2_mintime,stage2_maxtime)[[1]]

dat$anthro_bmi_stage1 <- select_oldest_measure("cbmi_",stage1_mintime,stage1_maxtime)[[1]]
dat$anthro_bmi_stage2 <- select_oldest_measure("cbmi_",stage2_mintime,stage2_maxtime)[[1]]
dat$anthro_bmi_stage3 <- select_oldest_measure("cbmi_",stage3_mintime,stage3_maxtime)[[1]]


dat$anthro_waist_stage1 <- select_oldest_measure("cabdo_",stage1_mintime,stage1_maxtime)[[1]]
dat$anthro_waist_stage2 <- select_oldest_measure("cabdo_",stage2_mintime,stage2_maxtime)[[1]]

dat$anthro_bmi_stage1_sds <- select_oldest_measure("czbmiwho06_",stage1_mintime,stage1_maxtime)[[1]]

dat$anthro_bmi_stage2_sds <- select_oldest_measure("czbmiwho06_",stage2_mintime,stage2_maxtime)[[1]]

dat$anthro_bmi_stage3_sds <- select_oldest_measure("czbmiwho06_",stage3_mintime,stage3_maxtime)[[1]]

#who categories of overweight and obesity
dat$anthro_overweightobese_stage1_binary <- NA
dat$anthro_overweightobese_stage1_binary[dat$anthro_bmi_stage1_sds>1] <-1
dat$anthro_overweightobese_stage1_binary[which(dat$anthro_bmi_stage1_sds>=(-2) &dat$anthro_bmi_stage1_sds<=1)] <-0

dat$anthro_obese_stage1_binary <- NA
dat$anthro_obese_stage1_binary[which(dat$anthro_bmi_stage1_sds>2)] <-1
dat$anthro_obese_stage1_binary[which(dat$anthro_bmi_stage1_sds<=1 &dat$anthro_bmi_stage1_sds>=(-2))] <-0

dat$anthro_overweightobese_stage2_binary <- NA
dat$anthro_overweightobese_stage2_binary[dat$anthro_bmi_stage2_sds>1] <-1
dat$anthro_overweightobese_stage2_binary[which(dat$anthro_bmi_stage2_sds>=(-2) &dat$anthro_bmi_stage2_sds<=1)] <-0

dat$anthro_obese_stage2_binary <- NA
dat$anthro_obese_stage2_binary[which(dat$anthro_bmi_stage2_sds>2)] <-1
dat$anthro_obese_stage2_binary[which(dat$anthro_bmi_stage2_sds<=1 &dat$anthro_bmi_stage2_sds>=(-2))] <-0

dat$anthro_overweightobese_stage3_binary <- NA
dat$anthro_overweightobese_stage3_binary[dat$anthro_bmi_stage3_sds>1] <-1
dat$anthro_overweightobese_stage3_binary[which(dat$anthro_bmi_stage3_sds>=(-2) &dat$anthro_bmi_stage3_sds<=1)] <-0

dat$anthro_obese_stage3_binary <- NA
dat$anthro_obese_stage3_binary[which(dat$anthro_bmi_stage3_sds>2)] <-1
dat$anthro_obese_stage3_binary[which(dat$anthro_bmi_stage3_sds<=1 &dat$anthro_bmi_stage3_sds>=(-2))] <-0

```

Fat mass index is not available for offspring

# Child's psychosocial & cognitive outcomes  {.tabset}

## Ideal variables

Ideal harmonised psychosocial and cognitive measures for offspring are:

*	Total behavioural difficulties (SDQ) - maternal report, continuous and dichotomised at >=17 [this is a combination of 4 subscales - excluding prosocial behaviour]
* Hyperactivity (SDQ) - maternal report, continuous and dichotomised at >=17
* Emotional symptoms (SDQ) - maternal report, continuous and dichotomised at >=5
* Conduct problems (SDQ) - maternal report, continuous and dichotomised at >=4
* Peer problems (SDQ) - maternal report, continuous and dichotomised at >=4
* Prosocial behaviour (SDQ)  - maternal report, continuous and dichotomised at <=6
* Internalising problems (SDQ) - maternal report, combination of emotional symptoms and peer problems
* Externalising problems (SDQ) - maternal report, combination of hyperactivity and conduct problems
* SCDC (Skuse Social Communication Score)
* Depressive symptoms (Moods and Feelings questionnaire - MFQ)
*	IQ and/or other or composite measures of intelligence/cognitive performance
*	Educational attainment

Ages are defined as:

* at birth or in first year of life (>0 to <1)
* childhood stage 1 (around age 2; >=1 to <3)
* childhood stage 2 (around age 4; >=3 to <5)
* childhood stage 3 (around age 6; >=5 to <8)
* childhood stage 4 (around age 9; >=8 to <11)

## Derive clean variables {.tabset}

Child's psychosocial & cognitive outcomes are not available 

# Child's immunological outcomes {.tabset}

## Ideal variables {.tabset}
Ideal harmonised immunological health measures for offspring are:

*	Doctor diagnosed asthma
* Wheeze
*	Doctor diagnosed eczema
* Eczema-like rash
*	Allergies (pet, pollen, mite/dust, bee/wasp, food)

## Derive clean variables {.tabset}

Child's immunological outcomes are not available

# Child's cardiometabolic health outcomes {.tabset}
## Ideal variables {.tabset}
 
* SBP
* HDL cholesterol
* TGs
* Glucose
* Total cholesterol
* Triglycerides
* Insulin
* Pro-insulin

## Derive clean variables {.tabset}

Child's cardiometabolic health outcomes are not available

# Covariates {.tabset}

## Ideal variables {.tabset}

**Pregnancy complications** 

* Hypertensive disorders of pregnancy (CH) 
* Gestational diabetes (CH)

**Demographics**

* Maternal age at conception
* Paternal age in pregnancy
* Maternal parity
* Both ethnicity
* Both education
* Both occupation

**Family relationships**

* Parents' marital status

**Parents'health**

* asthma
* eczema
* allergy
* stressful life events
* mental health disorder
* mother took supplements
* bmi
* drugs

**Child's health**

* Gestational age at delivery (and preterm binary) 
* Duration of breastfeeding (CH) 
* Child's physical activity (TC)  
* Child's own smoking (TC) 
* Child's own drinking of alcohol (TC) 
* Child's own consumption of caffeine (TC)
* Child's exposure to passive smoke (CH)

## Derive clean variables

Pregnancy complications

```{r eval=FALSE, include=TRUE}
#glycosuria, existing diabetes or gestational diabetes vs no glycosuria or diabetes
dat$covs_glycosuria_binary<-NA
dat$covs_glycosuria_binary[raw_dat$drvgesdiab=="No"&raw_dat$bkfdiabete=="No"]<-0
dat$covs_glycosuria_binary[raw_dat$drvgesdiab=="Yes"|raw_dat$bkfdiabete=="Yes"]<-1
dat$covs_glycosuria_binary[raw_dat$prnprexcont1dm=="Yes"]<-1
dat$covs_glycosuria_binary[raw_dat$prnprexcont2dm=="Yes"]<-1

# preeclampsia or pregnancy induced HT
dat$covs_hdp_binary<-NA
dat$covs_hdp_binary[raw_dat$bkfhyperpi=="No"&raw_dat$bkfpreeclm=="No"]<-0
dat$covs_hdp_binary[raw_dat$bkfhyperpi=="Mild to moderate"|raw_dat$bkfhyperpi== "Severe"|raw_dat$bkfhyperpi=="Yes, not classified"|raw_dat$bkfpreeclm=="Yes"]<-1
```

Demographics

```{r eval=FALSE, include=TRUE}
# gender
dat$covs_sex<-NA
dat$covs_sex[raw_dat$eclbabysex=="Male"]<-"male"
dat$covs_sex[raw_dat$eclbabysex=="Female"]<-"female"

# Maternal age 
# this was measured during pregnancy. I've used in years, but months is also available (agemm_mbagtt) 
dat$covs_age_mother_conception<-raw_dat$agemy_mbqall

# Paternal age
dat$covs_age_father_pregnancy<-raw_dat$agefy_fbqall

#Parity
dat$covs_parity_mother_binary<-NA
dat$covs_parity_mother_binary[raw_dat$eclregpart==0]<-0
dat$covs_parity_mother_binary[raw_dat$eclregpart>0]<-1

# Maternal ethnicity

dat$covs_ethnicity_mother<-NA
dat$covs_ethnicity_mother[raw_dat$eth0eth9gp%in%c("White British","White Other")]<-"white"
dat$covs_ethnicity_mother[raw_dat$eth0eth9gp%in%c("Mixed-White and Black", "Mixed-White and South Asian", "Other")]<-"mixed or other"
dat$covs_ethnicity_mother[raw_dat$eth0eth9gp%in%c("Pakistani", "Indian", "Bangladeshi")]<-"asian"
dat$covs_ethnicity_mother[raw_dat$eth0eth9gp%in%c("Black")]<-"black"

dat$covs_ethnicityBIB_mother <-NA
dat$covs_ethnicityBIB_mother[raw_dat$eth0eth9gp%in%c("White British","White Other")]<-"white european"
dat$covs_ethnicityBIB_mother[raw_dat$eth0eth9gp%in%c("Pakistani","Indian","Bangladeshi")]<-"south asian"
dat$covs_ethnicityBIB_mother[raw_dat$eth0eth9gp%in%c("Other","Mixed-White and Black","Mixed-White and South Asian","White Other","Black")]<-"other"

dat$covs_ethnicity_mother_binary <- NA
dat$covs_ethnicity_mother_binary[dat$covs_ethnicityBIB_mother == "white european"] <- "white"
dat$covs_ethnicity_mother_binary[dat$covs_ethnicityBIB_mother %in% c("other","south asian")] <- "not white"

# Paternal ethnicity

dat$covs_ethnicity_father<-NA
dat$covs_ethnicity_father[raw_dat$fbqethnicgrp%in%c("White")]<-"white"
dat$covs_ethnicity_father[raw_dat$fbqethnicgrp%in%c("Mixed", "Other")]<-"mixed or other"
dat$covs_ethnicity_father[raw_dat$fbqethnicgrp%in%c("Asian/Asian British","Chinese")]<-"asian"
dat$covs_ethnicity_father[raw_dat$fbqethnicgrp%in%c("Black/Black British")]<-"black"

dat$covs_ethnicity_father_binary <- NA
dat$covs_ethnicity_father_binary[dat$covs_ethnicity_father == "white"] <- "white"
dat$covs_ethnicity_father_binary[dat$covs_ethnicity_father %in% c("mixed or other","black","asian")] <- "not white"


# Maternal education (other/don't know/foreign unknown all coded as NA)
dat$covs_edu_mother<-NA
dat$covs_edu_mother[raw_dat$edu0mumede=="<5 GCSE equivalent"] <-0
dat$covs_edu_mother[raw_dat$edu0mumede=="5 GCSE equivalent"] <-1
dat$covs_edu_mother[raw_dat$edu0mumede=="A-level equivalent"] <-2
dat$covs_edu_mother[raw_dat$edu0mumede=="Higher than A-level"] <-3

# Paternal education
dat$covs_edu_father<-NA
dat$covs_edu_father[raw_dat$edu0fthede=="<5 GCSE equivalent"] <-0
dat$covs_edu_father[raw_dat$edu0fthede=="5 GCSE equivalent"] <-1
dat$covs_edu_father[raw_dat$edu0fthede=="A-level equivalent"] <-2
dat$covs_edu_father[raw_dat$edu0fthede=="Higher than A-level"] <-3

# Maternal occupation
# mother's occupation not recorded in the data currently in Bristol. Lifecycle have used variables 	bib6f06 (etc) from the BiB1000 sub study and all24e7 etc (from ALL-IN sub study). We have employment status, but this isn't the same as occupation and can't be used in its place.

# Paternal occupation (note the typos in raw_dat variables are real, not mistakes in the EPoCH code)
dat$covs_occup_father <- NA
dat$covs_occup_father[raw_dat$job0fthwrk %in% c("Modern professional occupations","Senior managers or administrators","Traditional professional occupations")] <- 3 #professional or managerial
dat$covs_occup_father[raw_dat$job0fthwrk %in% c("Clerical and intermediate occupations","Middle or junior managers")] <- 2 #skilled non-manual
dat$covs_occup_father[raw_dat$job0fthwrk %in% c("Technical and craft occupations","Semi-rountine manual and service occupations")] <- 1 #skilled manual
dat$covs_occup_father[raw_dat$job0fthwrk %in% c("Rountine manual and service occupations")] <- 0 #semi-skilled or unskilled

dat$covs_edu_father_highestlowest_binary <- NA
dat$covs_edu_father_highestlowest_binary[dat$covs_edu_father==0] <- 0
dat$covs_edu_father_highestlowest_binary[dat$covs_edu_father==3] <- 1

dat$covs_edu_mother_highestlowest_binary <- NA
dat$covs_edu_mother_highestlowest_binary[dat$covs_edu_mother==0] <- 0
dat$covs_edu_mother_highestlowest_binary[dat$covs_edu_mother==3] <- 1

dat$covs_occup_mother_highestlowest_binary <- NA
dat$covs_occup_mother_highestlowest_binary[dat$covs_occup_mother==0] <- 0
dat$covs_occup_mother_highestlowest_binary[dat$covs_occup_mother==3] <- 1

dat$covs_occup_father_highestlowest_binary <- NA
dat$covs_occup_father_highestlowest_binary[dat$covs_occup_father==0] <- 0
dat$covs_occup_father_highestlowest_binary[dat$covs_occup_father==3] <- 1
```

Family relationships

```{r eval=FALSE, include=TRUE}

# Living with partner
dat$covs_partner_lives_with_mother_prenatal<-NA
dat$covs_partner_lives_with_mother_prenatal[raw_dat$hhd0cohabt%in%c("Living with baby's father","Living with another partner")]<- 1
dat$covs_partner_lives_with_mother_prenatal[raw_dat$hhd0cohabt=="Not living with a partner"]<- 0

# mum married (or not separated)
dat$covs_married_mother_binary<-NA
dat$covs_married_mother_binary[raw_dat$hhd0martst%in%c("Married (1st marriage)","Re-married")]<-1
dat$covs_married_mother_binary[raw_dat$hhd0martst%in%c("Separated (still legally married","Divorced","Widowed","Single (never married)")]<-0


```

Parents' health
```{r eval=FALSE, include=TRUE}
#Mother's asthma (lots of missingness)
dat$covs_asthma_mother <- NA
dat$covs_asthma_mother[raw_dat$prnprexconasthma=="Yes"]<-1
dat$covs_asthma_mother[raw_dat$prnprexconasthma=="No"]<-0

#Mother's mental illness (lots of missingness)
dat$covs_mentalhealthdisorder_mother <- NA
dat$covs_mentalhealthdisorder_mother[raw_dat$prnprexconmentheal=="Yes"]<-1
dat$covs_mentalhealthdisorder_mother[raw_dat$prnprexconmentheal=="No"]<-0

#Maternal supplements during pregnancy (last 4 weeks)
dat$covs_supplements_mother_binary <-NA
dat$covs_supplements_mother_binary[raw_dat$vit0vitipr=="Yes"]<-1
dat$covs_supplements_mother_binary[raw_dat$vit0vitipr=="No"]<-0

#Drugs in pregnancy
dat$covs_drugs_mother_binary <-NA
dat$covs_drugs_mother_binary[raw_dat$drg0drguse=="Yes"]<-1
dat$covs_drugs_mother_binary[raw_dat$drg0drguse=="No"]<-0

# Maternal height
dat$covs_height_mother<-raw_dat$mms0height

# Paternal height
dat$covs_height_father<-raw_dat$fbqheight
dat$covs_height_father[dat$covs_height_father<quantile(dat$covs_height_father,na.rm=T,probs=0.01)|dat$covs_height_father>quantile(dat$covs_height_father,na.rm=T,probs=0.99)]<-NA

# Maternal weight
dat$covs_weight_mother<-raw_dat$mms0weight

# Paternal weight
dat$covs_weight_father<-raw_dat$fbqweight
dat$covs_weight_father[dat$covs_weight_father<quantile(dat$covs_weight_father,na.rm=T,probs=0.01)|dat$covs_weight_father>quantile(dat$covs_weight_father,na.rm=T,probs=0.99)]<-NA

# Maternal bmi
dat$covs_bmi_mother<-raw_dat$mms0mbkbmi

# Paternal bmi
dat$covs_bmi_father <- dat$covs_weight_father/(dat$covs_height_father/100)^2

```

Child's health
```{r eval=FALSE, include=TRUE}
dat$covs_gestage <- raw_dat$eclgestday/7

dat$covs_preterm_binary <- NA
dat$covs_preterm_binary[dat$covs_gestage<37]<-1
dat$covs_preterm_binary[dat$covs_gestage>=37]<-0

```

Other child's health variables are not available in the data we have in Bristol. Breastfeeding information was collected in a subcohort

# Negative control outcomes  {.tabset}
 * Pigeon infestation
 * Mouse infestation
 * lefthandness

Negative control variables are not available

# Genetic data {.tabset}

## GWAS info

**Alcohol and tobacco**

* Alcohol and tobacco SNPs were taken from Liu et al 2019 'Association Studies of Up to 1.2 Million Individuals Yield New Insights Into the Genetic Etiology of Tobacco and Alcohol Use'

Full summary statistics from all cohorts available from GSCAN.

There are 4 different exposures for smoking:

1)	Cigarettes per day. Categorical (5)
2)	Smoking initiation. Binary (Regular current V non smoker)
3)	Cessation. Binary (Regular current V former smoker)
4)	Age of initiation (Age first regularly smoked)

NB. The Smoking initiation one is problematic, and seems to be related to impulsive behaviour instead

**Caffeine**

* Caffeine SNPs were taken from Cornelis et al 2015 (same as Treur et al) 'Genome-wide meta-analysis identifies six (8 start) novel loci associated with habitual coffee consumption'

**Physical activity** 

* Physical activity SNPs were taken from Doherty et al 2018 'GWAS identifies 14 loci for device-measured physical activity and sleep duration'

* There are 2 different exposures for physical acitivity:
    1) Sedentary behaviour
    2) Overall activity

* PRS for physical activity were only derived for 'sedentary', there are SNPs related to 'overall activity' but the n(=2 after clumping) is so small and unable to make PRS. The SNP n for sedentary is also very small to note .

## Read in PRS data and clean

PRS created using Plink v1.9, then read in here:
```{r eval=FALSE, include=TRUE}
##nb I created the file labeled as 'mothers' but this holds both mums and child
prs_alcohol <- read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/final_scores/20210303_scores_alcohol_mothers.profile",
                              header = T,
                              check.names = F)

#combine FID and IID columns
prs_alcohol$id<-paste(prs_alcohol$FID,prs_alcohol$IID,sep="_")
#rename score for each exposure
prs_alcohol$prs_score_alcohol<-prs_alcohol$SCORE
## keep only the columns we want
prs_alcohol<-prs_alcohol[, which(names(prs_alcohol) %in% c("prs_score_alcohol","id"))]

### Smoking
#Smoking age initiation
prs_smoking_age_init <- read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/final_scores/20210324_scores_smoking_age_init.profile",
                          header = T,
                          check.names = F)

#combine FID and IID columns
prs_smoking_age_init$id<-paste(prs_smoking_age_init$FID,prs_smoking_age_init$IID,sep="_")
#rename score for each exposure
prs_smoking_age_init$prs_score_smoking_age_init<-prs_smoking_age_init$SCORE
## keep only the columns we want
prs_smoking_age_init<-prs_smoking_age_init[, which(names(prs_smoking_age_init) %in% c("prs_score_smoking_age_init","id"))]

#Smoking cessation
prs_smoking_cessation <- read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/final_scores/20210325_scores_smoking_cessation.profile",
                                   header = T,
                                   check.names = F)

#combine FID and IID columns
prs_smoking_cessation$id<-paste(prs_smoking_cessation$FID,prs_smoking_cessation$IID,sep="_")
#rename score for each exposure
prs_smoking_cessation$prs_score_smoking_cessation<-prs_smoking_cessation$SCORE
## keep only the columns we want
prs_smoking_cessation<-prs_smoking_cessation[, which(names(prs_smoking_cessation) %in% c("prs_score_smoking_cessation","id"))]

#smoking cigs per day
prs_smoking_cigs_pd <- read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/final_scores/20210325_scores_smoking_cigs_pd.profile",
                                    header = T,
                                    check.names = F)

#combine FID and IID columns
prs_smoking_cigs_pd$id<-paste(prs_smoking_cigs_pd$FID,prs_smoking_cigs_pd$IID,sep="_")
#rename score for each exposure
prs_smoking_cigs_pd$prs_score_smoking_cigs_pd<-prs_smoking_cigs_pd$SCORE
## keep only the columns we want
prs_smoking_cigs_pd<-prs_smoking_cigs_pd[, which(names(prs_smoking_cigs_pd) %in% c("prs_score_smoking_cigs_pd","id"))]

#Smoking initiation
prs_smoking_initiation <- read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/final_scores/20210325_scores_smoking_initiation.profile",
                                  header = T,
                                  check.names = F)

#combine FID and IID columns
prs_smoking_initiation$id<-paste(prs_smoking_initiation$FID,prs_smoking_initiation$IID,sep="_")
#rename score for each exposure
prs_smoking_initiation$prs_score_smoking_initiation<-prs_smoking_initiation$SCORE
## keep only the columns we want
prs_smoking_initiation<-prs_smoking_initiation[, which(names(prs_smoking_initiation) %in% c("prs_score_smoking_initiation","id"))]

#Caffeine
prs_caffeine <- read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/final_scores/20210325_scores_caffeine.profile",
                             header = T,
                             check.names = F)

#combine FID and IID columns
prs_caffeine$id<-paste(prs_caffeine$FID,prs_caffeine$IID,sep="_")
#rename score for each exposure
prs_caffeine$prs_score_caffeine<-prs_caffeine$SCORE
## keep only the columns we want
prs_caffeine<-prs_caffeine[, which(names(prs_caffeine) %in% c("prs_score_caffeine","id"))]

###Physical activity
## None for overall activity as too few variants to make PRS

#Sedentary

prs_sedentary <- read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/final_scores/20210326_scores_pa_sedentary.profile",
                                  header = T,
                                  check.names = F)
prs_sedentary$id<-paste(prs_sedentary$FID,prs_sedentary$IID,sep="_")
#rename score for each exposure
prs_sedentary$prs_score_sedentary<-prs_sedentary$SCORE
## keep only the columns we want
prs_sedentary<-prs_sedentary[, which(names(prs_sedentary) %in% c("prs_score_sedentary","id"))]
```

## Read in individual SNP data and clean

```{r eval=FALSE, include=TRUE}
#! Potential for duplicate snps to present in final dataset, as come from different snp lists for exposures - an option round this, is to rename with 'alc' 'smok' etc before name of snp??

snps_alcohol<-read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/individual_snp_score/alcohol_mothers_excluded_numb.raw", header=T)
snps_smoking_initiation<-read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/individual_snp_score/smoking_initiation_excluded_numb.raw", header=T)
snps_smoking_cigs_pd<-read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/individual_snp_score/smoking_cigs_pd_excluded_numb.raw", header=T)
snps_smoking_cessation<-read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/individual_snp_score/smoking_cessation_excluded_numb.raw", header=T)
snps_smoking_age_init<-read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/individual_snp_score/smoking_age_init_excluded_numb.raw", header=T)
snps_sedentary<-read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/individual_snp_score/pa_sedentary_excluded_numb.raw", header=T)
snps_overall_activity<-read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/individual_snp_score/pa_overall_activity_excluded_numb.raw", header=T)
snps_caffeine<-read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/genetic scores/individual_snp_score/caffeine_excluded_numb.raw", header=T)


snps_alcohol$id<-paste(snps_alcohol$FID,snps_alcohol$IID,sep="_")
## keep only the columns we want
snps_alcohol<-snps_alcohol[, -which(colnames(snps_alcohol) %in% c("FID","IID","SEX","PAT","MAT","PHENOTYPE"))]

snps_smoking_initiation$id<-paste(snps_smoking_initiation$FID,snps_smoking_initiation$IID,sep="_")
snps_smoking_initiation<-snps_smoking_initiation[, -which(colnames(snps_smoking_initiation) %in% c("FID","IID","SEX","PAT","MAT","PHENOTYPE"))]

snps_smoking_cigs_pd$id<-paste(snps_smoking_cigs_pd$FID,snps_smoking_cigs_pd$IID,sep="_")
snps_smoking_cigs_pd<-snps_smoking_cigs_pd[, -which(colnames(snps_smoking_cigs_pd) %in% c("FID","IID","SEX","PAT","MAT","PHENOTYPE"))]

snps_smoking_cessation$id<-paste(snps_smoking_cessation$FID,snps_smoking_cessation$IID,sep="_")
snps_smoking_cessation<-snps_smoking_cessation[, -which(colnames(snps_smoking_cessation) %in% c("FID","IID","SEX","PAT","MAT","PHENOTYPE"))]

snps_smoking_age_init$id<-paste(snps_smoking_age_init$FID,snps_smoking_age_init$IID,sep="_")
snps_smoking_age_init<-snps_smoking_age_init[, -which(colnames(snps_smoking_age_init) %in% c("FID","IID","SEX","PAT","MAT","PHENOTYPE"))]

snps_sedentary$id<-paste(snps_sedentary$FID,snps_sedentary$IID,sep="_")
snps_sedentary<-snps_sedentary[, -which(colnames(snps_sedentary) %in% c("FID","IID","SEX","PAT","MAT","PHENOTYPE"))]

snps_overall_activity$id<-paste(snps_overall_activity$FID,snps_overall_activity$IID,sep="_")
snps_overall_activity<-snps_overall_activity[, -which(colnames(snps_overall_activity) %in% c("FID","IID","SEX","PAT","MAT","PHENOTYPE"))]

snps_caffeine$id<-paste(snps_caffeine$FID,snps_caffeine$IID,sep="_")
snps_caffeine<-snps_caffeine[, -which(colnames(snps_caffeine) %in% c("FID","IID","SEX","PAT","MAT","PHENOTYPE"))]


```

## Quality control

PRS data
```{r eval=FALSE, include=TRUE}
#install.packages("readstata13")
library(readstata13)

exclude <- read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH GitHub/data_prep/cohorts/bib/pca_prep/directly_genotyped_released_ibd.genome.exclusions.uniq.txt")
exclude <- exclude$V1 #15628 to 14439
exclude_child <- read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH GitHub/data_prep/cohorts/bib/pca_prep/child/directly_genotyped_released_ibd.genome.exclusions.uniq.txt")
exclude_child <- exclude_child$V1 #13859

# EXCLUDE those on exclusion list - 
#mothers
prs_alcohol <- prs_alcohol[ ! prs_alcohol$id %in% exclude, ]
dim(prs_alcohol) 
#child 
prs_alcohol <- prs_alcohol[ ! prs_alcohol$id %in% exclude_child, ]
dim(prs_alcohol) 

#mothers
prs_smoking_age_init <- prs_smoking_age_init[ ! prs_smoking_age_init$id %in% exclude, ]
dim(prs_smoking_age_init) 
#child 
prs_smoking_age_init <- prs_smoking_age_init[ ! prs_smoking_age_init$id %in% exclude_child, ]
dim(prs_smoking_age_init)

#mothers
prs_smoking_cessation <- prs_smoking_cessation[ ! prs_smoking_cessation$id %in% exclude, ]
dim(prs_smoking_cessation) 
#child 
prs_smoking_cessation <- prs_smoking_cessation[ ! prs_smoking_cessation$id %in% exclude_child, ]
dim(prs_smoking_cessation) 

#mothers
prs_smoking_cigs_pd <- prs_smoking_cigs_pd[ ! prs_smoking_cigs_pd$id %in% exclude, ]
dim(prs_smoking_cigs_pd) 
#child 
prs_smoking_cigs_pd <- prs_smoking_cigs_pd[ ! prs_smoking_cigs_pd$id %in% exclude_child, ]
dim(prs_smoking_cigs_pd) 

#mothers
prs_smoking_initiation <- prs_smoking_initiation[ ! prs_smoking_initiation$id %in% exclude, ]
dim(prs_smoking_initiation)
#child 
prs_smoking_initiation <- prs_smoking_initiation[ ! prs_smoking_initiation$id %in% exclude_child, ]
dim(prs_smoking_initiation) 

#mothers
prs_caffeine <- prs_caffeine[ ! prs_caffeine$id %in% exclude, ]
dim(prs_caffeine)
#child 
prs_caffeine <- prs_caffeine[ ! prs_caffeine$id %in% exclude_child, ]
dim(prs_caffeine)

#mothers
prs_sedentary <- prs_sedentary[ ! prs_sedentary$id %in% exclude, ]
dim(prs_sedentary) 
#child 
prs_sedentary <- prs_sedentary[ ! prs_sedentary$id %in% exclude_child, ]
dim(prs_sedentary) 
```

Individual SNP data
```{r eval=FALSE, include=TRUE}

snps_alcohol <- snps_alcohol[ ! snps_alcohol$id %in% exclude, ]
dim(snps_alcohol) 
#child 
snps_alcohol <- snps_alcohol[ ! snps_alcohol$id %in% exclude_child, ]
dim(snps_alcohol) 

#mothers
snps_smoking_age_init <- snps_smoking_age_init[ ! snps_smoking_age_init$id %in% exclude, ]
dim(snps_smoking_age_init) 
#child 
snps_smoking_age_init <- snps_smoking_age_init[ ! snps_smoking_age_init$id %in% exclude_child, ]
dim(snps_smoking_age_init)

#mothers
snps_smoking_cessation <- snps_smoking_cessation[ ! snps_smoking_cessation$id %in% exclude, ]
dim(snps_smoking_cessation) 
#child 
snps_smoking_cessation <- snps_smoking_cessation[ ! snps_smoking_cessation$id %in% exclude_child, ]
dim(snps_smoking_cessation) 

#mothers
snps_smoking_cigs_pd <- snps_smoking_cigs_pd[ ! snps_smoking_cigs_pd$id %in% exclude, ]
dim(snps_smoking_cigs_pd) 
#child 
snps_smoking_cigs_pd <- snps_smoking_cigs_pd[ ! snps_smoking_cigs_pd$id %in% exclude_child, ]
dim(snps_smoking_cigs_pd) 

#mothers
snps_smoking_initiation <- snps_smoking_initiation[ ! snps_smoking_initiation$id %in% exclude, ]
dim(snps_smoking_initiation)
#child 
snps_smoking_initiation <- snps_smoking_initiation[ ! snps_smoking_initiation$id %in% exclude_child, ]
dim(snps_smoking_initiation) 

#mothers
snps_caffeine <- snps_caffeine[ ! snps_caffeine$id %in% exclude, ]
dim(snps_caffeine)
#child 
snps_caffeine <- snps_caffeine[ ! snps_caffeine$id %in% exclude_child, ]
dim(snps_caffeine)

#mothers
snps_sedentary <- snps_sedentary[ ! snps_sedentary$id %in% exclude, ]
dim(snps_sedentary) 
#child 
snps_sedentary <- snps_sedentary[ ! snps_sedentary$id %in% exclude_child, ]
dim(snps_sedentary) 

#mothers
snps_overall_activity <- snps_overall_activity[ ! snps_overall_activity$id %in% exclude, ]
dim(snps_overall_activity) 
#child 
snps_overall_activity <- snps_overall_activity[ ! snps_overall_activity$id %in% exclude_child, ]
dim(snps_overall_activity) 


```

## PCA
PCs made using Plink. Run mapping.R script and then bib_pc.sh/bib_pc_child.sh in Plink
```{r eval=FALSE, include=TRUE}
#Mum PCAs #7533
bib_pca <- read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/pca_prep/pcs/directly_genotyped_released_pca.eigenvec", header = TRUE)
bib_pca=subset(subset(bib_pca,select=-c(IID)))
bib_pca<-bib_pca%>%rename_with(tolower)
names(bib_pca)<-paste0("mum",names(bib_pca))
colnames(bib_pca)[1] <- "id"

## Child PCAs #6308
bib_pca_child <- read.table("/Users/ke14426/University of Bristol/grp-EPoCH - Documents/EPoCH Github/data_prep/cohorts/bib/pca_prep/pcs/child/directly_genotyped_released_pca.eigenvec", header = TRUE)
bib_pca_child=subset(subset(bib_pca_child,select=-c(IID)))

bib_pca_child<-bib_pca_child%>%rename_with(tolower)
names(bib_pca_child)<-paste0("child",names(bib_pca_child))
colnames(bib_pca_child)[1] <- "id"
```

## linking genetic data
```{r eval=FALSE, include=TRUE}
# Maternal

linkage <- read.csv("/Volumes/bib/phenotypic/released/2019-04-01/data/csv/Linkage_v7_gwaslp_Data.csv")
colnames(linkage)[5] <- "id"
link_maternal <- merge(linkage, bib_pca, by = "id")

#n =7533
link_maternal=subset(subset(link_maternal,select=-c(has_gwasmpreg,gwasrepmpreg)))
```

Read in linkage file  
```{r eval=FALSE, include=TRUE}
linkage_child <- read.csv("/Volumes/bib/phenotypic/released/2019-04-01/data/csv/Linkage_v7_gwaslc_Data.csv")
colnames(linkage_child)[6] <- "id"

## link to PCA
link_child <- merge(linkage_child, bib_pca_child, by = "id")
#n=5473

## select those only that have genotype linked with mothers
link_child <- subset(link_child, has_gwascpairm == 'has')
#n=5473
link_child=subset(subset(link_child,select=-c(has_gwasc,has_gwascpairm,gwasrepchild,has_gwascpairmpreg)))

```


Merge with genetic data
```{r eval=FALSE, include=TRUE}
#maternal and child PCs merge
genetic_merge <- merge(link_maternal,link_child[,-which(colnames(link_child) %in% c("id","MotherID","FatherID"))],by = c("PregnancyID"),all.x=TRUE,all.y=FALSE)
genetic_merge <- genetic_merge[, c(2,1,3,4,25,5:24,26:45)]


link_genetic <- left_join(genetic_merge, prs_alcohol, by = "id")
link_genetic <- left_join(link_genetic, prs_caffeine, by = "id")
link_genetic <- left_join(link_genetic, prs_smoking_age_init, by = "id")
link_genetic <- left_join(link_genetic, prs_smoking_cessation, by = "id")
link_genetic <- left_join(link_genetic, prs_smoking_cigs_pd, by = "id")
link_genetic <- left_join(link_genetic, prs_smoking_initiation, by = "id")
link_genetic <- left_join(link_genetic, prs_sedentary, by = "id")
#7533

#link_genetic <- left_join(bib_pca, link_child, by = "id")
link_genetic <- left_join(link_genetic, snps_alcohol, by = "id")
link_genetic <- left_join(link_genetic, snps_caffeine, by = "id")
link_genetic <- left_join(link_genetic, snps_smoking_age_init, by = "id")
link_genetic <- left_join(link_genetic, snps_smoking_cessation, by = "id")
link_genetic <- left_join(link_genetic, snps_smoking_cigs_pd, by = "id")
link_genetic <- left_join(link_genetic, snps_smoking_initiation, by = "id")
link_genetic <- left_join(link_genetic, snps_sedentary, by = "id")
link_genetic <- left_join(link_genetic, snps_overall_activity, by = "id")

#saveRDS(link_genetic,"/Volumes/data/bib/bib_all_genetic.rds")
```


Merge genetic and phenotypic data
```{r eval=FALSE, include=TRUE}
dat <- left_join(dat,bib_genetic_data, by = c("PregnancyID", "MotherID","ChildID","FatherID"), all.x = TRUE)
```

Z scores PRS
```{r eval=FALSE, include=TRUE}
prs_vars <- names(dat)[grep(names(dat),pattern="prs")]
scaled_prs <- as.data.frame(apply(dat[,prs_vars],2,scale))
names(scaled_prs) <- paste0(prs_vars,"_zscore")
library(dplyr)
dat <- bind_cols(dat,scaled_prs)

```

# Stratify by ethnicity

I have stratified by the two major ethnic groups in BiB: South Asian (n=5910) and white British (n=4545). This means 1058 pregnancies are excluded because they involve a mother from a different ethnic group, including mixed heritage south asian and white british. The other option is to stratify by white/all other ethnic groups, but it looks like most previous research involving BiB has stratified by south asian/white British.
```{r eval=FALSE, include=TRUE}
datwe <- dat[which(dat$covs_ethnicityBIB_mother=="white european"),]
datsa <- dat[which(dat$covs_ethnicityBIB_mother=="south asian"),]
```

## Create Z scores for each ethnic grouping separately

For stratified analyses

```{r eval=FALSE, include=TRUE}
#white european
datwe$anthro_birthweight_zscore <-scale(datwe$anthro_birthweight)
datwe$anthro_head_circ_birth_zscore <-scale(datwe$anthro_head_circ_birth)
datwe$anthro_crown_heel_birth_zscore <-scale(datwe$anthro_crown_heel_birth)

datwe$anthro_height_stage1_zscore <-scale(datwe$anthro_height_stage1)
datwe$anthro_height_stage2_zscore <-scale(datwe$anthro_height_stage2)
datwe$anthro_height_stage3_zscore <-scale(datwe$anthro_height_stage3)

datwe$anthro_weight_stage1_zscore <-scale(datwe$anthro_weight_stage1)
datwe$anthro_weight_stage2_zscore <-scale(datwe$anthro_weight_stage2)
datwe$anthro_weight_stage3_zscore <-scale(datwe$anthro_weight_stage3)

datwe$anthro_bmi_stage1_zscore <-scale(datwe$anthro_bmi_stage1)
datwe$anthro_bmi_stage2_zscore <-scale(datwe$anthro_bmi_stage2)
datwe$anthro_bmi_stage3_zscore <-scale(datwe$anthro_bmi_stage3)

datwe$anthro_head_circ_stage1_zscore <-scale(datwe$anthro_head_circ_stage1)
datwe$anthro_head_circ_stage2_zscore <-scale(datwe$anthro_head_circ_stage2)

datwe$anthro_waist_stage1_zscore <-scale(datwe$anthro_waist_stage1)
datwe$anthro_waist_stage2_zscore <-scale(datwe$anthro_waist_stage2)

datwe$covs_bmi_father_zscore <-scale(datwe$covs_bmi_father)
datwe$covs_bmi_mother_zscore <- scale(datwe$covs_bmi_mother)

#south asian
datsa$anthro_birthweight_zscore <-scale(datsa$anthro_birthweight)
datsa$anthro_head_circ_birth_zscore <-scale(datsa$anthro_head_circ_birth)
datsa$anthro_crown_heel_birth_zscore <-scale(datsa$anthro_crown_heel_birth)

datsa$anthro_height_stage1_zscore <-scale(datsa$anthro_height_stage1)
datsa$anthro_height_stage2_zscore <-scale(datsa$anthro_height_stage2)
datsa$anthro_height_stage3_zscore <-scale(datsa$anthro_height_stage3)

datsa$anthro_weight_stage1_zscore <-scale(datsa$anthro_weight_stage1)
datsa$anthro_weight_stage2_zscore <-scale(datsa$anthro_weight_stage2)
datsa$anthro_weight_stage3_zscore <-scale(datsa$anthro_weight_stage3)

datsa$anthro_bmi_stage1_zscore <-scale(datsa$anthro_bmi_stage1)
datsa$anthro_bmi_stage2_zscore <-scale(datsa$anthro_bmi_stage2)
datsa$anthro_bmi_stage3_zscore <-scale(datsa$anthro_bmi_stage3)

datsa$anthro_head_circ_stage1_zscore <-scale(datsa$anthro_head_circ_stage1)
datsa$anthro_head_circ_stage2_zscore <-scale(datsa$anthro_head_circ_stage2)

datsa$anthro_waist_stage1_zscore <-scale(datsa$anthro_waist_stage1)
datsa$anthro_waist_stage2_zscore <-scale(datsa$anthro_waist_stage2)

datsa$covs_bmi_father_zscore <-scale(datsa$covs_bmi_father)
datsa$covs_bmi_mother_zscore <- scale(datsa$covs_bmi_mother)

# all together

dat$anthro_birthweight_zscore <-scale(dat$anthro_birthweight)
dat$anthro_head_circ_birth_zscore <-scale(dat$anthro_head_circ_birth)
dat$anthro_crown_heel_birth_zscore <-scale(dat$anthro_crown_heel_birth)

dat$anthro_height_stage1_zscore <-scale(dat$anthro_height_stage1)
dat$anthro_height_stage2_zscore <-scale(dat$anthro_height_stage2)
dat$anthro_height_stage3_zscore <-scale(dat$anthro_height_stage3)

dat$anthro_weight_stage1_zscore <-scale(dat$anthro_weight_stage1)
dat$anthro_weight_stage2_zscore <-scale(dat$anthro_weight_stage2)
dat$anthro_weight_stage3_zscore <-scale(dat$anthro_weight_stage3)

dat$anthro_bmi_stage1_zscore <-scale(dat$anthro_bmi_stage1)
dat$anthro_bmi_stage2_zscore <-scale(dat$anthro_bmi_stage2)
dat$anthro_bmi_stage3_zscore <-scale(dat$anthro_bmi_stage3)

dat$anthro_head_circ_stage1_zscore <-scale(dat$anthro_head_circ_stage1)
dat$anthro_head_circ_stage2_zscore <-scale(dat$anthro_head_circ_stage2)

dat$anthro_waist_stage1_zscore <-scale(dat$anthro_waist_stage1)
dat$anthro_waist_stage2_zscore <-scale(dat$anthro_waist_stage2)

dat$covs_bmi_father_zscore <-scale(dat$covs_bmi_father)
dat$covs_bmi_mother_zscore <- scale(dat$covs_bmi_mother)
```

# Save files {.tabset}
You must be connected to the server: smb://rdsfcifs.acrc.bris.ac.uk/MRC-IEU-research/
```{r eval=FALSE, include=TRUE}
saveRDS(dat,"/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/bib/bib_all_pheno.rds")
saveRDS(datsa,"/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/bib/bib_sa_pheno.rds")
saveRDS(datwe,"/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/bib/bib_we_pheno.rds")
```

# Short cut to working on this dataset
```{r eval=FALSE, include=TRUE}
dat <- readRDS("/Volumes/MRC-IEU-research/projects/ieu2/p5/015/working/data/bib/bib_all_pheno.rds")
```
